README.md

```diff
@@ -247,11 +247,223 @@ Unlike with local kernels, Hydrogen does not kill remote kernels when it disconn
+
+- Run your container mapping the port of the gateway
+- Give your container a name
+- Make it run an infinite loop that just keeps the container alive:
+
+```bash
+docker run -d -p 8888:8888 --name hydro  hydro bash -c "while true; do sleep 10; done"
+```
+
+**Note**: you will only be able to run one container using that port mapping. So, if you had another container using that port, you will have to stop that one first. Or alternatively, you can create a mapping to a new port and add that configuration in the Hydrogen settings (see below).
+
+- Execute an interactive bash session in your running container:
+
+```bash
+docker exec -it hydro bash
+```
+
+- From that interactive session, start the gateway
+- Specify the IP `0.0.0.0` to make your container listen to public connections
+- Specify the port that you exposed in your `Dockerfile`:
+
+```bash
+jupyter kernelgateway --ip=0.0.0.0 --port=8888
+```
+
+#### Run Docker Container with Docker Compose
+
+- Create a `docker-compose.yml` file with something like:
+
+```yml
+version: '2'
+services:
+  hydro:
+    build: .
+    ports:
+      - "8888:8888"
+    command: bash -c "while true; do sleep 10; done"
+```
+
+- The `docker-compose.yml` file has a port mapping using the port exposed in the `Dockerfile` and used in the `jupyter kernelgateway` command
+- The `command` overrides the default `CMD` in the `Dockerfile` (if there was one) and executes an infinite loop that would just keep the container alive
+
+**Note**: you will only be able to run one container using that port mapping. So, if you had another container using that port, you will have to stop that one first. Or alternatively, you can create a mapping to a new port and add that configuration in the Hydrogen settings (see below).
+
+- Now start (and build) your container with `docker-compose`:
+
+```bash
+docker-compose up -d
+```
+
+- Check the name of your running container with:
+
+```bash
+docker-compose ps
+```
+
+- Execute an interactive bash session in your running container (use the name from above), e.g.:
+
+```bash
+docker exec -it myproject_hydro_1 bash
+```
+
+- From that interactive session, start the gateway
+- Specify the IP `0.0.0.0` to make your container listen to public connections
+- Specify the port that you exposed in your `Dockerfile`:
+
+```bash
+jupyter kernelgateway --ip=0.0.0.0 --port=8888
+```
+
+#### Connect Atom
+
+- Go to the settings in Atom with: `ctrl-shift-p` and type `Settings View: Open`
+- Go to the "Packages" section
+- Type `Hydrogen` and go to package settings
+- In the section "List of kernel gateways to use" add settings for the container your created
+- Use a `name` that you can remind when running Hydrogen
+- In the `baseUrl` section use the host or IP that you use to access your Docker containers:
+  - If you are using Docker Toolbox in Windows or Mac (or Docker for Windows, Docker for Mac), as it will be running in a virtual machine, the IP (host) would probably be like: `192.168.99.100`
```

Comment:
If they use anything different than Docker in Linux it will be running on a virtual machine (Docker for Windows, Docker for Mac or Docker Toolbox). So it would be a local virtual network IP. And if everything goes normally, it would be precisely that IP: `192.168.99.100`. But that could vary in each installation.

In any case, if someone got to the point in where he is using Docker in a non-Linux system, he most probably knows that. Also it is [officially documented here](https://docs.docker.com/machine/get-started/#/run-containers-and-experiment-with-machine-commands).

But if it's required, I could add a note stating that one possible way to get the IP of a locally running Docker in a non-Linux system could be: `docker-machine ip default`. Although this won't necessarily work "all" the time, for example if someone changed the name of the Docker Machine from `default` to something different. But most of the time (in a local environment) it would work.

And in another case, if the user is running Docker in a host different than the one running Atom, maybe a different machine in the same network or a (*not recommended*) public VPS on the cloud, the IP would be that of that other machine. And that would depend on each user.